storage-proofs/post/src/fallback/vanilla.rs

```diff
@@ -332,9 +342,33 @@ impl<'a, Tree: 'a + MerkleTreeTrait> ProofScheme<'a> for FallbackPoSt<'a, Tree>
                             challenge_index,
                         )?;
 
-                        tree.gen_cached_proof(challenged_leaf_start as usize, Some(rows_to_discard))
+                        let proof = tree.gen_cached_proof(
+                            challenged_leaf_start as usize,
+                            Some(rows_to_discard),
+                        );
+                        let valid = if let Ok(proof) = &proof {
```

Comment:
Only FYI (it is correct). I would write this as:

```rust
let valid = match &proof {
    Ok(proof) => {
        proof.validate(challenged_leaf_start as usize)
            && proof.root() == priv_sector.comm_r_last
    }
    Err(_) => false
}
```

I generally use `if let` only if there is no `else`, in all other cases a `match`.

Perhaps I'd even go as far and write the whole part as:

```rust
match &proof {
    Ok(proof) => {
        if proof.validate(challenged_leaf_start as usize)
            && proof.root() == priv_sector.comm_r_last {
            Ok(ProofOrFault::Proof(proof))
        } else {
            Ok(ProofOrFault::Fault(sector_id))
        }
    }
    Err(_) => Ok(ProofOrFault::Fault(sector_id))
}
```

Reply:
Agreed, this case (though not necessarily about always eschewing `else`).